ui/pages/snap-account-redirect/components/snap-account-redirect-context.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import {
+  BannerAlert,
+  BannerAlertSeverity,
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Text,
+} from '../../../components/component-library';
+import { SnapAccountRedirectProps } from '../snap-account-redirect';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import SnapAccountRedirectMessage from './snap-account-redirect-message';
+
+const SnapAccountRedirectContent = ({
+  url,
+  snapName,
+  isBlockedUrl,
+  message,
+}: SnapAccountRedirectProps) => {
+  const t = useI18nContext();
+  const learnMoreAboutBlockedUrls =
+    'https://support.metamask.io/hc/en-us/articles/4428045875483--Deceptive-site-ahead-when-trying-to-connect-to-a-site';
+
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      justifyContent={JustifyContent.spaceBetween}
+      paddingTop={4}
+    >
+      <Box
+        gap={4}
+        paddingLeft={2}
+        paddingRight={2}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        alignItems={AlignItems.center}
+      >
+        <Text
+          data-testid="snap-account-redirect-content-title"
+          textAlign={TextAlign.Center}
+          variant={TextVariant.headingLg}
+        >
+          {t('snapAccountRedirectFinishSigningTitle')}
+        </Text>
+        {isBlockedUrl ? (
+          <Box display={Display.Flex} paddingLeft={4} paddingRight={4}>
+            <BannerAlert severity={BannerAlertSeverity.Danger}>
+              <Text>
+                {t('snapUrlIsBlocked', [
+                  <Button
+                    variant={ButtonVariant.Link}
+                    size={ButtonSize.Inherit}
+                    onClick={() =>
+                      global.platform.openTab({
+                        url: learnMoreAboutBlockedUrls,
+                      })
+                    }
+                    key={`snap-url-is-blocked-learn-more-button`}
+                  >
+                    {t('learnMore')}
+                  </Button>,
+                ])}
+              </Text>
+            </BannerAlert>
+          </Box>
+        ) : null}
+        <Text
+          data-testid="snap-account-redirect-content-description"
+          textAlign={TextAlign.Center}
+          variant={TextVariant.bodyMd}
+        >
+          {t('snapAccountRedirectSiteDescription', [snapName])}
+        </Text>
+        {url !== '' || message !== '' ? (
+          <SnapAccountRedirectMessage
```

Comment:
I assume this shouldn't render when `isBlocked`? That doesn't seem to be the case

Reply:
based on these designs we still render the message and/or url when it is blocked but we also render a banner alert. One point of confusion here might be that the `SnapAccountRedirectMessage` component also renders the URL.
<img width="950" alt="Screenshot 2023-10-11 at 11 41 28 AM" src="https://github.com/MetaMask/metamask-extension/assets/22918444/1c71f840-a9c9-4182-93c5-cffbbd799843">

Reply:
Are we sure about showing blocked URLs to the user? 🤔 

The snaps team is about to ship a feature to allow links in custom UIs and we were planning to not show URLs if they are in the phishing list.

Reply:
Do we have a link to these designs?

Reply:
@eriknson https://www.figma.com/file/K0csZegZ5XQ6TjfKk3n4Z8/Add-dialog-to-inform-about-redirection-to-snap-site-%2335?type=design&node-id=604-2586&mode=design

Reply:
cc @raulmono

Reply:
Thanks for the heads up – not displaying the URL when it's blocked sounds good. It might look like this: 
![image](https://github.com/MetaMask/metamask-extension/assets/11038018/5e381650-f0d3-4972-88d0-3b411a03163f)

Reply:
@raulmono would we still want to show the `Go to site` button?

Reply:
@owencraston yeah, you're right – we should show that either

Reply:
wdyt? @eriknson @FrederikBolding

Reply:
Sounds good to me 👍

Reply:
Updated to reflect that change
<img width="385" alt="Screenshot 2023-10-11 at 3 22 58 PM" src="https://github.com/MetaMask/metamask-extension/assets/22918444/d1bf45be-621a-4aa6-801c-943adfe23627">